app/app.go

```diff
@@ -308,3 +317,30 @@ func (a *app) submitFileTransferAuditlog(ctx context.Context, userID string, dev
 	}
 	return nil
 }
+
+func (a *app) Shutdown(ctx context.Context, timeout time.Duration) error {
+	ticker := time.NewTicker(timeout / time.Duration(len(a.shutdownCancels)+1))
+	for _, cancel := range a.shutdownCancels {
+		cancel()
+		<-ticker.C
+	}
+	<-ticker.C
```

Comment:
I don't understand why the handler blocks on the ticker when exiting the loop. Did you mean to stop the ticker?
```suggestion
	ticker.Stop()
```
```suggestion
```

Reply:
Because of the +1, you have an extra "tick" to consume to make the shutdown complete in the grace period.